libs/base/ui/structure/text/text.component.ts

```diff
@@ -11,6 +11,76 @@ export class TextComponent extends LitElement implements TextProperties {
 
   protected render(): TemplateResult | void {
     if (!this.content) return;
-    return html`${unsafeHTML(this.content)}`;
+    const c = this.appendStyleToHeaders(this.content);
+    return html`${unsafeHTML(c)}`;
+  }
+
+  /**
+   * Modifies an HTML string by adding custom inline styles to specific HTML elements
+   * based on their tags, classes, and screen sizes. The styles align roughly with the
+   * Oryx design system without reusing the design system components directly.
+   *
+   * @param htmlString - The HTML string to be modified.
+   * @returns The modified HTML string with custom inline styles.
+   */
+  protected appendStyleToHeaders(htmlString: string): string {
```

Comment:
I'm just wondering about efficiency of such a solution. 

The component is basically generating custom (non lit-template) html on it's own, now with additional preprocessing. If it would be only for the (fairly static) content, then all good. But we intend to use it as a low level component for components like product title. In a consequence, we are generating html in a custom way also there, as this approach kind of encourages it 🤔